community/kernel/src/main/java/org/neo4j/kernel/impl/store/record/SchemaRuleDeserializer2_0to3_1.java

```diff
@@ -138,6 +138,13 @@ public static ConstraintRule readUniquenessConstraintRule( long id, int labelId,
                 readOwnedIndexRule( buffer ) );
     }
 
+    public static ConstraintRule readNodeKeyConstraintRule( long id, int labelId, ByteBuffer buffer )
+    {
+        return new ConstraintRule( id,
+                ConstraintDescriptorFactory.nodeKeyForLabel( labelId, readConstraintPropertyKeys( buffer ) ),
+                readOwnedIndexRule( buffer ) );
+    }
```

Comment:
Where is this used? This class is about deserialising legacy schema rule records, I think.

Also, I don't know why this isn't already the case, but the `read*` methods should assign their result to a local variable, before passing the values into any other methods. Otherwise, a refactoring that changes the parameter order will end up (almost certainly unintentionally) implying a format change.